parsing/parser.mly

```diff
@@ -2745,8 +2746,10 @@ fun_body:
   | fun_seq_expr
       { Pfunction_body $1 }
 ;
-%inline match_cases:
-  xs = preceded_or_separated_nonempty_llist(BAR, match_case)
+match_cases:
+    xs = preceded_or_separated_nonempty_llist(BAR, match_case) %prec below_BAR
+    { xs }
+  | LPAREN xs = match_cases RPAREN
```

Comment:
I'd drop support for nested delimiters, which doesn't seem to add anything useful.

Reply:
I put them in for consistency with other places you can use parentheses, I agree that they are not useful and would be fine with removing them.